plugins/copy-to-clipboard/index.html

```diff
@@ -20,19 +20,171 @@
 
 <section>
 	<h1>How to use</h1>
-	<p>In addition to including the plugin file with your PrismJS build, ensure <a href="https://clipboardjs.com/">Clipboard.js</a> is loaded before the plugin.</p>
 
-	<p>The simplest way to include Clipboard.js is to use any of the
-		<a href="https://github.com/zenorocha/clipboard.js/wiki/CDN-Providers">recommended CDNs</a>. If you're using Browserify, Clipboard.js will be loaded automatically
+	<p>The plugin depends on <a href="https://clipboardjs.com/">clipboard.js</a> and the Prism <a href="https://prismjs.com/plugins/toolbar/">Toolbar</a> plugin. In addition to including the plugin file with your PrismJS build, ensure they are loaded before the plugin.</p>
```

Comment:
Do we normally mention internal dependencies like this? Should we going forward? If we do, maybe we should link to the explanation of how that works in the main page?

Reply:
> internal dependencies

Toolbar or clipboard.js? 

Copy to clipboard will automatically load clipboard.js, so it's essentially just an implementation detail. I don't think it should be mentioned here.

As for Toolbar, all of Prism's components can have dependencies. We have Autoloader for languages but we don't offer an automatic way to load plugins right now. People that load plugins from CDN's by including files have to know this.

Reply:
> Toolbar or clipboard.js?

The Toolbar.

> Copy to clipboard will automatically load clipboard.js, so it's essentially just an implementation detail. I don't think it should be mentioned here.

lol I was actually thinking the opposite. We don't include explanations of required internal dependencies in our docs generally because (imo) our expectation is they'll use one of the tools (website download or babel plugin) to generate their bundle. As an example, the previewers plugin [requires css-extras](https://github.com/PrismJS/prism/blob/master/components.json#L1398) but we don't mention it in the [docs](https://github.com/PrismJS/prism/blob/master/plugins/previewers/index.html). Clipboard.js is an external dependency which wouldn't be resolved by either of those approaches so imo should be mentioned.

Reply:
> our expectation is they'll use one of the tools (website download or babel plugin)

> As an example, the previewers plugin requires css-extras but we don't mention it in the docs.

True but for plugins, we only have download and babel plugin. None work in combination with CDNs.

The best way to solve this might be to make Autoloader support loading plugins. We don't have to mention dependencies if Autoloader will handle them for our users.

> Clipboard.js is an external dependency which wouldn't be resolved by either of those approaches so imo should be mentioned.

But nobody has to resolve it in the first place because [the plugin itself will load clipboard.js](https://github.com/PrismJS/prism/blob/fd1081d256f2355b1c6f3690ea50282a9a907502/plugins/copy-to-clipboard/prism-copy-to-clipboard.js#L34). Why mention something nobody has to worry about?

Reply:
> The best way to solve this might be to make Autoloader support loading plugins.

I'm open to this, although the autoloader functions by checking the language. We'd have to add a setting or data attribute to indicate plugins to load.

> But nobody has to resolve it in the first place because the plugin itself will load clipboard.js. Why mention something nobody has to worry about?

Because loading the plugin remotely is bad for performance compared to bundling it with the plugin, nor is it clear that using this plugin means it will be automatically loading additional scripts from external locations, which could have security implications. For both of these reasons, I definitely think it should be mentioned here.

Reply:
I also want to be clear that this discussion shouldn't block the merge (once the Russian text is validated).

Reply:
> security implications

True. That alone is enough to keep it.

> We'd have to add a setting or data attribute to indicate plugins to load.

That was my first idea as well. I'll make an issue.

> this discussion shouldn't block the merge

Definitely. 

---

Also, coming back to the original question: Do we want to keep the mentioning of Toolbar?

I don't have any hard feelings on that subject. I said that people need this information but we haven't had any people making issues about it yet, so it might not be a problem? Not sure.

Reply:
I'm ok with leaving it.

Reply:
Me too.